lib/validators/url_validator.rb

```diff
@@ -17,18 +18,31 @@ def url_present?(record)
   end
 
   def url_starts_with_protocol?(record)
-    error = 'Url does not start with http or https'
+    uri = Addressable::URI.parse(record.url)
+    return false if uri.blank?
 
-    if record.url !~ /^https?/
+    error = 'Url does not start with http, https, or ftp'
+
+    if %w(http https ftp).exclude? uri.scheme
       create_validation_error(record, error)
     else
       true
     end
   end
 
+  def encoded?(original_url)
+    decoded = Addressable::URI.unescape(original_url)
```

Comment:
`Addressable::URI` has `unencode` which is aliased to `unescape`. Perhaps we should use "encode" consistently?

Reply:
Yup, can do. I wonder why it doesn't have `decode` though

Reply:
@andrewgarner Done